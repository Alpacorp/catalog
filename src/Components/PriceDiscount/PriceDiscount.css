.price-discount {
  text-decoration: line-through;
}
