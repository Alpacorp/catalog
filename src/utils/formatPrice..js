export const formatPrice = new Intl.NumberFormat('es-CO', {
  style: 'currency',
  currency: 'COP',
  minimumFractionDigits: 0
});