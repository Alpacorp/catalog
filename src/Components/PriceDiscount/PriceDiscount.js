import "./PriceDiscount.css";

const PriceDiscount = ({ price, formatPrice }) => {
  return (
    <div>
      <h3 className="price-discount" aria-label={price} tabIndex="0">
        {formatPrice.format(price)}
      </h3>
    </div>
  );
};

export default PriceDiscount;
