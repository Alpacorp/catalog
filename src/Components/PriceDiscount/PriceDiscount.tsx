import PropTypes from "prop-types";
import "./PriceDiscount.css";

interface PriceDiscountProps {
  price: number;
  formatPrice: any;
}

const PriceDiscount: React.FC<PriceDiscountProps> = ({
  price,
  formatPrice,
}) => {
  return (
    <div>
      <h3 className="price-discount" tabIndex={0}>
        {formatPrice.format(price)}
      </h3>
    </div>
  );
};

PriceDiscount.propTypes = {
  price: PropTypes.number.isRequired,
  formatPrice: PropTypes.object.isRequired,
};

export default PriceDiscount;
