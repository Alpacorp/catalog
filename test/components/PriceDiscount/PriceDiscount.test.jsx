import { render, screen } from "@testing-library/react";
import PriceDiscount from "../../../src/Components/PriceDiscount/PriceDiscount";
import { formatPrice } from "../../../src/utils/formatPrice.";
import propTypes from "prop-types";

describe("Test on <PriceDiscount />", () => {
  const price = 100;
  const className = "price-discount";
  const ariaLabel = price;
  const tabIndex = 0;

  test("the component should render to match snapshot", () => {
    render(<PriceDiscount price={price} formatPrice={formatPrice} />);
    expect(
      <PriceDiscount price={price} formatPrice={formatPrice} />
    ).toMatchSnapshot();
  });

  test("the component should have H3 tag", () => {
    render(<PriceDiscount price={price} formatPrice={formatPrice} />);
    expect(screen.getByRole("heading", { level: 3 })).toBeTruthy();
  });

  test('The component should have in H3 tag "price-discount" className and "price" aria-label and "tab-index" properties', () => {
    render(<PriceDiscount price={price} formatPrice={formatPrice} />);
    expect(screen.getByRole("heading", { level: 3 }).className).toBe(className);
    expect(
      screen.getByRole("heading", { level: 3 }).getAttribute("aria-label")
    ).toBe(ariaLabel.toString());
    expect(
      screen.getByRole("heading", { level: 3 }).getAttribute("tabIndex")
    ).toBe(tabIndex.toString());
  });

  test("The component should have two props", () => {
    render(<PriceDiscount price={price} formatPrice={formatPrice} />);
    expect(PriceDiscount.propTypes).toEqual({
      price: propTypes.number.isRequired,
      formatPrice: propTypes.object.isRequired,
    });
    expect(Object.keys(PriceDiscount.propTypes).length).toBe(2);
    // console.log(Object.keys(PriceDiscount.propTypes)[0]);
  });
});
