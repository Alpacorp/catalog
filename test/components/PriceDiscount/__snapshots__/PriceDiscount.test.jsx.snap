// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <PriceDiscount /> the component should render to match snapshot 1`] = `
<PriceDiscount
  formatPrice={NumberFormat {}}
  price={100}
/>
`;
